### PR TITLE
B7 alpha

### PR DESCRIPTION
**Pull request by hzoo**
I guess also v7.0.0-alpha.3?

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -108,7 +108,7 @@ it('should list used helpers in file.babel', function (cb) {
 
 it('should not rename ignored files', function (cb) {
 	var stream = babel({
-		ignore: /fixture/
+		ignore: [/fixture/]
```
Comment (hzoo) [inline test.js]: breaking change in babel-core was requiring ignore to be an array

File: index.js [hunk 2]
```diff
@@ -36,17 +35,19 @@ module.exports = function (opts) {
 
 			var res = babel.transform(file.contents.toString(), fileOpts);
 
-			if (file.sourceMap && res.map) {
-				res.map.file = replaceExtension(res.map.file);
-				applySourceMap(file, res.map);
-			}
+			if (res !== null) {
```
Comment (hzoo) [inline index.js]: breaking change is that ignored files return `null`?

File: index.js [hunk 3]
```diff
@@ -26,7 +25,7 @@ module.exports = function (opts) {
 		}
 
 		try {
-			var fileOpts = objectAssign({}, opts, {
+			var fileOpts = Object.assign({}, opts, {
```
Comment (hzoo) [inline index.js]: dropping node 0.10/0.12 so removing this for native

File: index.js (hunk 2 above)
Comment (xtuc) [inline index.js]: Yes, some user will get `code` of null.

File: index.js [hunk 4]
```diff
@@ -26,27 +25,29 @@ module.exports = function (opts) {
...
+				if (file.sourceMap && res.map) {
+					res.map.file = replaceExtension(res.map.file);
+					applySourceMap(file, res.map);
+				}
 
-			if (!res.ignored) {
-				file.contents = new Buffer(res.code);
-				file.path = replaceExtension(file.path);
-			}
+				if (!res.ignored) {
```
Comment (loganfsmyth) [inline index.js]: I don't think we need this `if` since the `null` check does this now.

**Pull request merged**